core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/UndirectedInfraBuilder.java

```diff
@@ -267,6 +269,58 @@ private Set<RJSLoadingGaugeType> getCompatibleGaugeTypes(RJSLoadingGaugeType tra
         };
     }
 
+    /** Computes the curves RangeMap of a track section for both directions. */
+    private EnumMap<Direction, RangeMap<Double, Double>> makeCurves(RJSTrackSection track) {
+        var curves = new EnumMap<Direction, RangeMap<Double, Double>>(Direction.class);
+        for (var dir : Direction.values()) {
+            var newMap = TreeRangeMap.<Double, Double>create();
+            if (dir == Direction.FORWARD)
+                newMap.putCoalescing(Range.closed(0., track.length), 0.);
+            else
+                newMap.putCoalescing(Range.closed(-track.length, 0.), 0.);
+            curves.put(dir, newMap);
+        }
+
+        if (track.curves != null) {
+            for (var rjscurve : track.curves) {
+                // Because of calling order in makeTrackSection, the curve should already be simplified
+                // and not have any boundary issue worth throwing an exception.
+                curves.get(Direction.FORWARD)
+                        .putCoalescing(Range.closed(rjscurve.begin, rjscurve.end), rjscurve.radius);
+                curves.get(Direction.BACKWARD)
+                        .putCoalescing(Range.closed(-rjscurve.end, -rjscurve.begin), rjscurve.radius);
```

Comment:
The positions being negative here isn't consistent with the rest of the project, I wouldn't have expected it